### ISSUE TITLE
Some training from stronger networks apparently helps (eg ELF)

### ISSUE THREAD

**Issue by Friday9i** [gcp/leela-zero#1780]
And what is significantly stronger than current LZ@1600 visits?
Hummm, maybe LZ@6400 visits for example (which is better than ELFv1@1600 visits)... 
Hence, instead of 20% of ELF games, why not include 10% or 20% of deeper LZ games? Without "no resign" games, in order to avoid very long games and to limit the total time used for them.
And that could be used all along, even for 40b : -).

**Comment by Friday9i** [comment]
Yeah, seems a good idea: it's a bit slower but as strong as ELFv1, and it will probably be better to train 40b nets (as we saw with 20b: without 20b selfplay, 20b nets didn't progress well).

**Comment by john45678** [comment]
I would love to know where we are compared to AZ. This is, for me, the most frustrating thing about DM not, at least, continuing to play games with AZ.
(sorry, not intending to hijack your thread).

**Comment by alphaladder** [comment]
just promote to 40b directly  when 20b exceeds elf.

**Comment by Crazyeight101** [comment]
@Friday9i at this point, I'm questioning whether this is actually true or not.  You would anticipate ELFv1 and 20b self-play to increase the quality of the 40bs that are being trained off of the LZ training data set, as they supposedly should be stronger than the ELFv0 and 15b data, and yet 40bs trained using data past LZ157 either play at about the same level as or slightly weaker than the 40b 157 net.  They also seem to play better against the 20b and ELFv1 nets than the 40b 157 net does, though, so the results are mixed.

I think there's something different going on here than "training data produced from stronger networks helps."  I don't know what it is, though. =P

**Comment by Friday9i** [comment]
@Crazyeight101 regarding your "going back to higher visits", that's not what I'm saying!
I think 1600 visits are more than enough for selfplay (and probably 800 visits should be enough), just that a reasonable % of deeper games should help. That has the advantage of 1) being fast overall 2) to get deep games to help train

**Comment by OmnipotentEntity** [comment]
Because ELF has problems with ladders, and ELF is trained on 800 playouts (approx 1600 visits), I am somewhat worried that official Leela Zero weights will "forget" about ladders.  Having a few % of deep reading games could prevent that.

**Comment by jokkebk** [comment]
Just a stupid question: Doesn't small visit count actually help keeping ladders in memory -- that way the self-play games will start playing losing ladders and provide many data points where this leads to bad results. Whereas if the reading is deep enough, no losing ladders occur?

Maybe on the server side where modified MCTS is used to train the network at higher visits it would notice the move leading to ladder should be avoided, but in training games it doesn't matter if the ladders are "realized" or just "avoided"? I haven't read the AZ paper that well though so I might be utterly wrong.

**Comment by MartinDevelopment** [comment]
> Doesn't small visit count actually help keeping ladders in memory 

For the network to not play out a bad ladder it has to play it out entirely or avoid it. Therefore you either need a lot of visits or a network that avoids ladders.

**Comment by roy7** [comment]
LZ learned to play ladders out to check them by giving ladders a high prior. ELF learned to avoid ladders completely by giving them a low prior. I assume this is because LZ had higher playouts in training to work with.
